### PR TITLE
Restructured registry packages

### PR DESCRIPTION
**Pull request by yueh**
Moved larger registry together with their related classes instead of putting unrelated classes into the same package.

### CODE REVIEW

File: src/main/java/appeng/core/features/registries/inscriber/InscriberRecipe.java [hunk 1]
```diff
@@ -57,7 +57,7 @@
 	@Nonnull
 	private final InscriberProcessType type;
 
-	public InscriberRecipe( @Nonnull final Collection<ItemStack> inputs, @Nonnull final ItemStack output, @Nullable final ItemStack top, @Nullable final ItemStack bot, @Nonnull final InscriberProcessType type )
+	InscriberRecipe( @Nonnull final Collection<ItemStack> inputs, @Nonnull final ItemStack output, @Nullable final ItemStack top, @Nullable final ItemStack bot, @Nonnull final InscriberProcessType type )
```
Comment (orod-org) [inline src/main/java/appeng/core/features/registries/inscriber/InscriberRecipe.java]: MINOR "inputs" is marked "javax.annotation.Nonnull" but is not initialized in this constructor. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2637)
MINOR "maybeTop" is marked "javax.annotation.Nonnull" but is not initialized in this constructor. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2637)
MINOR "maybeBot" is marked "javax.annotation.Nonnull" but is not initialized in this constructor. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2637)
MINOR "type" is marked "javax.annotation.Nonnull" but is not initialized in this constructor. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2637)
MINOR "output" is marked "javax.annotation.Nonnull" but is not initialized in this constructor. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2637)

File: src/main/java/appeng/core/features/registries/inscriber/InscriberRecipe.java [hunk 2]
```diff
@@ -57,7 +57,7 @@
...
 	{
 		this.inputs = new ArrayList<ItemStack>( inputs.size() );
```
Comment (orod-org) [inline src/main/java/appeng/core/features/registries/inscriber/InscriberRecipe.java]: MINOR Replace the type specification in this constructor call with the diamond operator ("<>"). [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2293)

**Comment by orod-org** [comment]
SonarQube analysis reported 21 issues
* CRITICAL 6 critical
* MAJOR 2 major
* MINOR 13 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [GrinderRecipeManager.java#L95](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/6097c6f868e5fee642f38ea974e571431214661b/src/main/java/appeng/core/features/registries/grinder/GrinderRecipeManager.java#L95): Define a constant instead of duplicating this literal "Null is not accepted as input itemstack." 3 times. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. CRITICAL [GrinderRecipeManager.java#L96](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/6097c6f868e5fee642f38ea974e571431214661b/src/main/java/appeng/core/features/registries/grinder/GrinderRecipeManager.java#L96): Define a constant instead of duplicating this literal "Null is not accepted as output itemstack." 3 times. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. CRITICAL [GrinderRecipeManager.java#L97](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/6097c6f868e5fee642f38ea974e571431214661b/src/main/java/appeng/core/features/registries/grinder/GrinderRecipeManager.java#L97): Define a constant instead of duplicating this literal "Turns must be > 0" 3 times. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. CRITICAL [GrinderRecipeManager.java#L194](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/6097c6f868e5fee642f38ea974e571431214661b/src/main/java/appeng/core/features/registries/grinder/GrinderRecipeManager.java#L194): The Cyclomatic Complexity of this method "oreRegistered" is 13 which is greater than 10 authorized. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. CRITICAL [GrinderRecipeManager.java#L196](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/6097c6f868e5fee642f38ea974e571431214661b/src/main/java/appeng/core/features/registries/grinder/GrinderRecipeManager.java#L196): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1067)
1. CRITICAL [InscriberRegistry.java#L175](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/6097c6f868e5fee642f38ea974e571431214661b/src/main/java/appeng/core/features/registries/inscriber/InscriberRegistry.java#L175): The Cyclomatic Complexity of this method "build" is 12 which is greater than 10 authorized. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. MAJOR [GrinderRecipeManager.java#L347](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/6097c6f868e5fee642f38ea974e571431214661b/src/main/java/appeng/core/features/registries/grinder/GrinderRecipeManager.java#L347): Method new appeng.core.features.registries.grinder.GrinderRecipeManager$CacheKey(ItemStack) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [InscriberRegistry.java#L119](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/6097c6f868e5fee642f38ea974e571431214661b/src/main/java/appeng/core/features/registries/inscriber/InscriberRegistry.java#L119): Add a constructor to the class. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1258)
1. MINOR [CellRegistry.java#L41](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/6097c6f868e5fee642f38ea974e571431214661b/src/main/java/appeng/core/features/registries/cell/CellRegistry.java#L41): Replace the type specification in this constructor call with the diamond operator ("<>"). [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2293)
1. MINOR [InscriberRegistry.java#L53](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/6097c6f868e5fee642f38ea974e571431214661b/src/main/java/appeng/core/features/registries/inscriber/InscriberRegistry.java#L53): Replace the type specification in this constructor call with the diamond operator ("<>"). [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2293)

**Comment by yueh** [comment]
Test this please.

File: src/main/java/appeng/recipes/handlers/Inscribe.java [hunk 3]
```diff
@@ -63,7 +64,12 @@ public void register() throws RegistrationError, MissingIngredientError
 		final ItemStack output = this.getOutput().getItemStack();
 		final InscriberProcessType type = InscriberProcessType.INSCRIBE;
 
-		final IInscriberRecipe recipe = new InscriberRecipe( inputs, output, top, bot, type );
+		final IInscriberRecipe recipe = builder.withInputs( inputs )
```
Comment (orod-org) [inline src/main/java/appeng/recipes/handlers/Inscribe.java]: MINOR Parameter 1 to this call is marked "javax.annotation.Nonnull" but null is passed. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2637)
MINOR Parameter 1 to this call is marked "javax.annotation.Nonnull" but null is passed. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2637)

File: src/main/java/appeng/recipes/handlers/Inscribe.java [hunk 4]
```diff
@@ -63,7 +64,12 @@ public void register() throws RegistrationError, MissingIngredientError
...
+				.withOutput( output )
+				.withTopOptional( top )
```
Comment (orod-org) [inline src/main/java/appeng/recipes/handlers/Inscribe.java]: MAJOR Null passed for non-null parameter of appeng.api.features.IInscriberRecipeBuilder.withTopOptional(ItemStack) in appeng.recipes.handlers.Inscribe.register() [rule](https://sonar.orod.org/coding_rules#rule_key=findbugs%3ANP_NULL_PARAM_DEREF)

File: src/main/java/appeng/recipes/handlers/Press.java [hunk 5]
```diff
@@ -63,7 +64,12 @@ public void register() throws RegistrationError, MissingIngredientError
 		final ItemStack output = this.getOutput().getItemStack();
 		final InscriberProcessType type = InscriberProcessType.PRESS;
 
-		final IInscriberRecipe recipe = new InscriberRecipe( inputs, output, top, bot, type );
+		final IInscriberRecipe recipe = builder.withInputs( inputs )
+				.withOutput( output )
+				.withTopOptional( top )
+				.withBottomOptional( bot )
```
Comment (orod-org) [inline src/main/java/appeng/recipes/handlers/Press.java]: MAJOR Null passed for non-null parameter of appeng.api.features.IInscriberRecipeBuilder.withBottomOptional(ItemStack) in appeng.recipes.handlers.Press.register() [rule](https://sonar.orod.org/coding_rules#rule_key=findbugs%3ANP_NULL_PARAM_DEREF)

File: src/main/java/appeng/tile/misc/TileInscriber.java [hunk 6]
```diff
@@ -403,18 +404,28 @@ public IInscriberRecipe getTask()
 				final List<ItemStack> inputs = Lists.newArrayList( startingItem );
 				final InscriberProcessType type = InscriberProcessType.INSCRIBE;
 
-				return new InscriberRecipe( inputs, renamedItem, plateA, plateB, type );
+				final IInscriberRecipeBuilder builder = AEApi.instance().registries().inscriber().builder();
+				return builder.withInputs( inputs )
+						.withOutput( renamedItem )
+						.withTopOptional( plateA )
+						.withBottomOptional( plateB )
```
Comment (orod-org) [inline src/main/java/appeng/tile/misc/TileInscriber.java]: MAJOR Null passed for non-null parameter of appeng.api.features.IInscriberRecipeBuilder.withBottomOptional(ItemStack) in appeng.tile.misc.TileInscriber.getTask() [rule](https://sonar.orod.org/coding_rules#rule_key=findbugs%3ANP_NULL_PARAM_DEREF)

**Comment by orod-org** [comment]
SonarQube analysis reported 32 issues
* CRITICAL 6 critical
* MAJOR 7 major
* MINOR 19 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [GrinderRecipeManager.java#L95](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/6097c6f868e5fee642f38ea974e571431214661b/src/main/java/appeng/core/features/registries/grinder/GrinderRecipeManager.java#L95): Define a constant instead of duplicating this literal "Null is not accepted as input itemstack." 3 times. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. CRITICAL [GrinderRecipeManager.java#L96](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/6097c6f868e5fee642f38ea974e571431214661b/src/main/java/appeng/core/features/registries/grinder/GrinderRecipeManager.java#L96): Define a constant instead of duplicating this literal "Null is not accepted as output itemstack." 3 times. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. CRITICAL [GrinderRecipeManager.java#L97](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/6097c6f868e5fee642f38ea974e571431214661b/src/main/java/appeng/core/features/registries/grinder/GrinderRecipeManager.java#L97): Define a constant instead of duplicating this literal "Turns must be > 0" 3 times. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1192)
1. CRITICAL [GrinderRecipeManager.java#L194](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/6097c6f868e5fee642f38ea974e571431214661b/src/main/java/appeng/core/features/registries/grinder/GrinderRecipeManager.java#L194): The Cyclomatic Complexity of this method "oreRegistered" is 13 which is greater than 10 authorized. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. CRITICAL [GrinderRecipeManager.java#L196](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/6097c6f868e5fee642f38ea974e571431214661b/src/main/java/appeng/core/features/registries/grinder/GrinderRecipeManager.java#L196): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1067)
1. CRITICAL [InscriberRegistry.java#L175](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/6097c6f868e5fee642f38ea974e571431214661b/src/main/java/appeng/core/features/registries/inscriber/InscriberRegistry.java#L175): The Cyclomatic Complexity of this method "build" is 12 which is greater than 10 authorized. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. MAJOR [GrinderRecipeManager.java#L347](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/6097c6f868e5fee642f38ea974e571431214661b/src/main/java/appeng/core/features/registries/grinder/GrinderRecipeManager.java#L347): Method new appeng.core.features.registries.grinder.GrinderRecipeManager$CacheKey(ItemStack) appears to call the same method on the same object redundantly [rule](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. MAJOR [InscriberRegistry.java#L119](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/6097c6f868e5fee642f38ea974e571431214661b/src/main/java/appeng/core/features/registries/inscriber/InscriberRegistry.java#L119): Add a constructor to the class. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1258)
1. MINOR [CellRegistry.java#L41](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/6097c6f868e5fee642f38ea974e571431214661b/src/main/java/appeng/core/features/registries/cell/CellRegistry.java#L41): Replace the type specification in this constructor call with the diamond operator ("<>"). [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2293)
1. MINOR [InscriberRegistry.java#L53](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/6097c6f868e5fee642f38ea974e571431214661b/src/main/java/appeng/core/features/registries/inscriber/InscriberRegistry.java#L53): Replace the type specification in this constructor call with the diamond operator ("<>"). [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2293)

**Pull request merged**